src/ServiceProvider.php

```diff
@@ -15,7 +15,7 @@ public function register()
     {
         // Register the main class to use with the facade
         $this->app->bind(BaseFactory::class, function () {
-            return new Factory;
+            return new Factory($app->make(Dispatcher::class));
```

Comment:
Using `$app`in this way will not work, because we are in a closure and that variable is even not passed to the closure too. You should edit this, and pass `$app` to the callback `[... function ($app)/.. ]`

Reply:
Also, you should modify the composer.json, because this functionality is only provided from the v8 of the framework